Failure on x86_64-linux [(full log)](https://logs.nix.gsc.io/?key=nixos/nixpkgs.34582&attempt_id=none)

<details><summary>Partial log (click to expand)</summary><p>

```
cc1plus: warning: unrecognized command line option '-Wno-deprecated-register'
make[3]: *** [Makefile:863: Libxml2Parser.lo] Error 1
make[3]: Leaving directory '/tmp/nix-build-squid-4.0.23.drv-0/squid-4.0.23/src/esi'
make[2]: *** [Makefile:6942: all-recursive] Error 1
make[2]: Leaving directory '/tmp/nix-build-squid-4.0.23.drv-0/squid-4.0.23/src'
make[1]: *** [Makefile:5961: all] Error 2
make[1]: Leaving directory '/tmp/nix-build-squid-4.0.23.drv-0/squid-4.0.23/src'
make: *** [Makefile:582: all-recursive] Error 1
builder for ‘/nix/store/235bjmqd5q83vlm0jqa7g5wri6x0y6qd-squid-4.0.23.drv’ failed with exit code 2
error: build of ‘/nix/store/08xp24bkr3rhcbmqhhll7flziwsh0a26-squid-3.5.27.drv’, ‘/nix/store/235bjmqd5q83vlm0jqa7g5wri6x0y6qd-squid-4.0.23.drv’ failed
```
</p></details>

--- a/pkgs/servers/squid/4.nix
+++ b/pkgs/servers/squid/4.nix
@@ -13,6 +13,10 @@ stdenv.mkDerivation rec {
     perl openldap pam db cyrus_sasl libcap expat libxml2 openssl
   ];
 
+  prePatch = ''
+    substituteInPlace configure --replace "/usr/local/include/libxml2" "${libxml2.dev}/include/libxml2"
+  '';
+
   configureFlags = [
     "--enable-ipv6"
     "--disable-strict-error-checking"

--- a/pkgs/servers/squid/4.nix
+++ b/pkgs/servers/squid/4.nix
@@ -2,11 +2,11 @@
 , expat, libxml2, openssl }:
 
 stdenv.mkDerivation rec {
-  name = "squid-4.0.21";
+  name = "squid-4.0.23";
 
   src = fetchurl {
     url = "http://www.squid-cache.org/Versions/v4/${name}.tar.xz";
-    sha256 = "0cwfj3qpl72k5l1h2rvkv1xg0720rifk4wcvi49z216hznyqwk8m";
+    sha256 = "0a8g0zs3xayfkxl8maq823b14lckvh9d5lf7ryh9rx303xh1mdqq";
   };
 
   buildInputs = [

--- a/pkgs/servers/squid/default.nix
+++ b/pkgs/servers/squid/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, fetchurl, perl, openldap, pam, db, cyrus_sasl, libcap
+{ stdenv, fetchurl, fetchpatch, perl, openldap, pam, db, cyrus_sasl, libcap
 , expat, libxml2, openssl }:
 
 stdenv.mkDerivation rec {
@@ -11,6 +11,19 @@ stdenv.mkDerivation rec {
 
   buildInputs = [
     perl openldap pam db cyrus_sasl libcap expat libxml2 openssl
+  ];
+
+  patches = [
+    (fetchpatch {
+      name = "CVE-2018-1000024.patch";
+      url = http://www.squid-cache.org/Versions/v3/3.5/changesets/SQUID-2018_1.patch;
+      sha256 = "0vzxr4rmybz0w4c1hi3szvqawbzl4r4b8wyvq9vgq1mzkk5invpg";
+    })
+    (fetchpatch {
+      name = "CVE-2018-1000027.patch";
+      url = http://www.squid-cache.org/Versions/v3/3.5/changesets/SQUID-2018_2.patch;
+      sha256 = "1a8hwk9z7h1j0c57anfzp3bwjd4pjbyh8aks4ca79nwz4d0y6wf3";
+    })
   ];
 
   configureFlags = [

